security/middleware.py

```diff
@@ -30,14 +29,19 @@ class BaseMiddleware(object):
 
     def load_setting(self, setting, value):
         """
-        Called initially for each of the keys in REQUIRED_SETTINGS and OPTIONAL_SETTINGS,
-        and again whenever any of these settings change (from the setting_changed signal).
-        Passed the setting key and the new value, which may be None for the keys in
-        OPTIONAL_SETTINGS. If no setting keys are defined then this method is never called.
+        Called initially for each of the keys in REQUIRED_SETTINGS and
+        OPTIONAL_SETTINGS, and again whenever any of these settings change
+        (from the setting_changed signal). Passed the setting key and the new
+        value, which may be None for the keys in OPTIONAL_SETTINGS. If no
+        setting keys are defined then this method is never called.
         """
+        raise NotImplementedError()
 
     def _on_setting_changed(self, sender, setting, value, **kwargs):
-        if setting in self.REQUIRED_SETTINGS or setting in self.OPTIONAL_SETTINGS:
+        if (
+            setting in self.REQUIRED_SETTINGS
```

Comment:
Still not sure if I like this style. Any reason the or in on the second line, rather than the first. The clauses would line up nicer if the and/or/etc ended the lines.